test/extensions/filters/http/ext_proc/filter_test.cc

```diff
@@ -1980,7 +1980,49 @@ TEST_F(HttpFilterTest, ProcessingModeResponseHeadersOnlyWithoutCallingDecodeHead
 
 // Using the default configuration, verify that the "clear_route_cache_ flag makes the appropriate
 // callback on the filter.
-TEST_F(HttpFilterTest, ClearRouteCache) {
+TEST_F(HttpFilterTest, ClearRouteCacheBodyMutation) {
```

Comment:
I had both of those tests to show that body modification doesn't trigger the clearing.
ClearRouteCacheUnchanged does no header or body modification.
ClearRouteCacheHeaderMutation does only header modification.
ClearRouteCacheBodyMutation does only body modification.
ClearRouteCacheDiasallowed does header modification while disabling the cache clearing (fixed this).

I could remove the body modification test because it doesn't check any extra code paths from the unchanged one.

With the new changes I have:
ClearRouteCacheUnchanged: does no header or body modification, expecting no cache clearing callbacks.
ClearRouteCacheHeaderMutation: does only header modification, expecting cache clearing callbacks..
ClearRouteCacheDiasallowedHeaderMutation: does header modification while disabling the cache clearing, expecting no cache clearing callbacks.